tensorflow/lite/tools/optimize/quantize_model.cc

```diff
@@ -811,37 +855,44 @@ TfLiteStatus QuantizeSharedRange(ModelT* model, ErrorReporter* error_reporter) {
 TfLiteStatus QuantizeWeightsInputOutput(
     ModelT* model, bool allow_float,
     const std::unordered_set<string>& operator_names,
-    ErrorReporter* error_reporter) {
+    const TensorType& activations_type, ErrorReporter* error_reporter) {
   for (size_t subgraph_idx = 0; subgraph_idx < model->subgraphs.size();
        subgraph_idx++) {
     SubGraphT* subgraph = model->subgraphs.at(subgraph_idx).get();
     for (size_t op_idx = 0; op_idx < subgraph->operators.size(); op_idx++) {
       OperatorT* op = subgraph->operators[op_idx].get();
       const BuiltinOperator op_code =
           model->operator_codes[op->opcode_index]->builtin_code;
-      operator_property::OperatorProperty property =
-          GetOperatorProperty(operator_names, model, subgraph_idx, op_idx,
-                              subgraph->tensors[op->outputs[0]]->name);
-
-      if (!property.quantizable && !allow_float) {
-        TF_LITE_REPORT_ERROR(error_reporter,
-                             "Quantization not yet supported for op: %s",
-                             EnumNameBuiltinOperator(op_code));
+      operator_property::OperatorProperty property = GetOperatorProperty(
+          operator_names, model, subgraph_idx, op_idx,
+          subgraph->tensors[op->outputs[0]]->name, activations_type);
+
+      if (activations_type == TensorType_INT16 && !property.quantizable &&
+          !allow_float) {
+        error_reporter->Report(
+            "Quantization to 16x8-bit not yet supported for op: %s",
+            EnumNameBuiltinOperator(op_code));
+        return kTfLiteError;
+      } else if (!property.quantizable && !allow_float) {
+        error_reporter->Report("Quantization not yet supported for op: %s",
```

Comment:
`TF_LITE_REPORT_ERROR should be used instead of error_reporter, so log strings can be removed to save space if needed.`